tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (C) 2018-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from openvino.tools.mo.front.common.replacement import FrontReplacementPattern
+from openvino.tools.mo.front.common.partial_infer.utils import int64_array
+from openvino.tools.mo.front.tf.graph_utils import create_op_node_with_second_input
+from openvino.tools.mo.graph.graph import Graph, rename_node
+from openvino.tools.mo.ops.GRUCell import GRUCell
+from openvino.tools.mo.ops.concat import Concat
+from openvino.tools.mo.ops.split import AttributedSplit
+from openvino.tools.mo.ops.transpose import Transpose
+
+
+class GRUBlockCellToGRUCell(FrontReplacementPattern):
+
+    enabled = True
+
+    def find_and_replace_pattern(self, graph: Graph):
+        for tf_gru_block_cell in graph.get_op_nodes(op='GRUBlockCell'):
+            original_name = tf_gru_block_cell.soft_get('name', tf_gru_block_cell.id)
+            tf_gru_block_cell['name'] = original_name + '/to_be_removed'
+
+            new_gru_cell = GRUCell(graph, {}).create_node()
+            rename_node(new_gru_cell, original_name)
+
+            tf_gru_block_cell.in_port(0).get_connection().set_destination(new_gru_cell.in_port(0))
+            tf_gru_block_cell.in_port(1).get_connection().set_destination(new_gru_cell.in_port(1))
+
+            concat_w = Concat(graph, {'name': original_name + '/Concat_W',
```

Comment:
I think the Concat->Split->Concat sequence can be simplified to Split(W_ru) followed by Concat({W_u, W_r, W_c})

Reply:
Good point, I've applied the suggestion in a separate PR https://github.com/openvinotoolkit/openvino/pull/12078